src/main/scala/strawman/collection/SortedMap.scala

```diff
@@ -26,6 +26,12 @@ trait SortedMapOps[K, +V, +CC[X, Y] <: SortedMap[X, Y] with SortedMapOps[X, Y, C
   def flatMap[K2, V2](f: ((K, V)) => IterableOnce[(K2, V2)])(implicit ordering: Ordering[K2]): CC[K2, V2] =
     sortedMapFromIterable(View.FlatMap(coll, f))
 
+  def collect[K2, V2](pf: PartialFunction[(K, V), (K2, V2)])(implicit ordering: Ordering[K2]): CC[K2, V2] =
+    flatMap { (kv: (K, V)) =>
+      if (pf.isDefinedAt(kv)) View.Single(pf(kv))
+      else View.Empty
+    }
```

Comment:
This works for me on 2.12 and 2.13. A failure in overload resolution could be due to https://github.com/scala/bug/issues/10194 but the fix for it is present in our currently used PR snapshot and in 2.13.0-M1. The original implementation in https://github.com/scala/scala/pull/5307 was based on Dotty. Maybe it has a similar bug?